File: library/src/main/java/org/fbme/lib/iec61499/declarations/EventAssociation.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.iec61499.Reference;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface EventAssociation {

    @Nullable EventDeclaration getEvent();

    @NotNull Reference<ParameterDeclaration> getParameterReference();

    void remove();
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/EventAssociation.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.ContainedElement
import org.fbme.lib.common.Reference

interface EventAssociation : ContainedElement {
    override val container: EventDeclaration?
    val parameterReference: Reference<ParameterDeclaration>
}
